### ISSUE TITLE
Container overflows with some extensions

### ISSUE THREAD

**Issue by N6REJ** [N6REJ/embryo#38]
_From @N6REJ on November 11, 2014 9:4_

![right-overflow](https://cloud.githubusercontent.com/assets/1850089/4990682/544779e6-6950-11e4-96ed-75616969e51f.jpg)

for extensions that don't follow pure BS3 rules, we need to set the limit to how wide they can go to prevent overflow

_Copied from original issue: N6REJ/ae-template#1_

**Comment by N6REJ** [comment]
_From @MaxOnTheHill on November 11, 2014 11:53_

wow ... weird kunena wouldn't have a max-wodth on it's main layout elements to be 100% of the containing element???

Last time I used Kunenan I do know I had to add a few extra css rules ... but thought they'd have fixed that with their latest releases??

I would have thought the container element was not the element that was wrapping outside ... but rather the Kunena element that was pushing outside the 100% width ... as container is defined within bootstrap itself ... and kunena should contain itself within whichever element it is placed?

Just a thought ... as applying a style to the container element ... isn't the element we're wanting to control is it?

Perhaps it's something more like the com_content area ... but again ... should be the extension wrapper we contain to only 100% max width?

**Comment by N6REJ** [comment]
well, we have container limiting the view-able area by its very function.  but that makes it the outer border so you might be right, content-area might be better, I'll give that a try shortly.
Kunena hasn't had a update in a VERY long time.  I'm beginning to think its a dead project.  They were working on a responsive version ( https://github.com/Kunena/Kunena-Forum ) 
In the Live AlterEgo template I had to do quite a bit of work to make kunena behave.  I could, supply a .css specific to this project.  I'll reopen this and take a closer look with fresh eyes.  I think about 0400 things started to blur together

**Comment by N6REJ** [comment]
ok, so because Kunena use's "container" as part of its class we either have to use 
#content-area .container {
max-width: 100%;
}
or 
.container { max-width: 100%; }
I'm leaning towards doing it in the container itself as any extension that calls the container class will have the same problem regardless of where its located.  This would fix it on a wider scale then doing so in content-area only.

**Comment by N6REJ** [comment]
@MaxOnTheHill after a lot of thought and review I'm firmly convinced its in the best interest to have the max-width in the container element.  There could be countless other extensions that use the .container class and actually makes sense for them to and irregardless of where they are or what they do we NEVER want to allow them to overflow.  Setting a max-width is a 100% safe way to accomplish this and actually helps insure things behave as intended by the template.  Here are 2 images showing the behavior.
after max-width
![after](https://cloud.githubusercontent.com/assets/1850089/5190821/bc78f0b2-74ae-11e4-88cc-37a51175c12c.JPG)
before max-width
![before](https://cloud.githubusercontent.com/assets/1850089/5190820/bc77a680-74ae-11e4-86f3-e1b582255ad0.JPG)

The borders are just to help illustrate the behavior so its clearer whats occurring.  Honestly I don't understand the opposition to this fix.

**Comment by N6REJ** [comment]
fixed in commit

**Comment by MaxOnTheHill** [comment]
yep ... max-width: 100%; should probably be the case for ALL elements I would think ... and I know I have added it to site builds at times with:
* {max-width:100%;}

PS @N6REJ : good to see your making progress with this template build Troy, I hope you get some good things coming your way for the work done ;)
